src/Manifolds.jl

```diff
@@ -435,68 +435,119 @@ function shortest_geodesic(M::Manifold, x, y, T::AbstractVector)
+tangent vector of `v` as well as the sectional curvatures `κ` required for
+computing Jacobi fields, i.e. the first value (along `v`) is zero.
+"""
+tangent_orthonormal_basis(M::Manifold,x,v) = error("A tangent orthonormal basis not implemented on $(typeof(M)) at $(typeof(x)) with first direction $(typeof(v)).")
+
+"""
+    AbstractVectorTransportMethod
+
+An abstract type, ehich method to use for a vector transport, i.e. within
+[`vector_transport_to`](@ref), [`vector_transport_direction`](@ref) or
+[`vector_transport_along`](@ref). 
+"""
+abstract type AbstractVectorTransportMethod end
+
+"""
+    ParallelTransport <: AbstractVectorTransportMethod
+
+Specify to use parallel transport as vector transport method within
+[`vector_transport_to`](@ref), [`vector_transport_direction`](@ref) or
+[`vector_transport_along`](@ref). 
+"""
+struct ParallelTransport <: AbstractVectorTransportMethod end
+
+"""
+    ProjectTangent <: AbstractVectorTransportMethod
+
+Specify to use projection onto tangent space as vector transport method within
+[`vector_transport_to`](@ref), [`vector_transport_direction`](@ref) or
+[`vector_transport_along`](@ref). See [`project_tangent`](@ref) for details.
+"""
+struct ProjectTangent <: AbstractVectorTransportMethod end
+
+"""
+    vector_transport_to!(M::Manifold, vto, x, v, y [, m])
 
 Vector transport of vector `v` at point `x` to point `y`. The result is saved
-to `vto`. By default, [`project_tangent!`](@ref) is used but this may change in
-the future.
+to `vto`. By default, the method `m` is [`ParallelTransport`](@ref).
 """
-vector_transport_to!(M::Manifold, vto, x, v, y) = project_tangent!(M, vto, y, v)
+vector_transport_to!(M::Manifold, vto, x, v, y) = vector_transport_to!(M,vto,x,v,y,ParallelTransport())
+vector_transport_to!(M::Manifold, vto, x, v, y, m::ProjectTangent) = project_tangent!(M, vto, y, v)
+
+function vector_transport_to!(M::Manifold, vto, x, v, y, m::AbstractVectorTransportMethod)
+    error("vector transport from a point of type $(typeof(x)) to a type $(typeof(y)) on a $(typeof(M)) for a vector of type $(v) and the $(typeof(m)) not yet implemented.")
+end
+
 
 """
-    vector_transport_to(M::Manifold, x, v, y)
+    vector_transport_to(M::Manifold, x, v, y[,method=::ParallelTransport])
 
-Vector transport of vector `v` at point `x` to point `y`.
+Vector transport of vector `v` at point `x` to point `y` using the method `m`,
+which defaults to [`ParallelTransport`](@ref).
 """
-function vector_transport_to(M::Manifold, x, v, y)
+vector_transport_to(M::Manifold, x, v, y) = vector_transport_to(M,x,v,y,ParallelTransport())
+function vector_transport_to(M::Manifold, x, v, y, m)
     vto = similar_result(M, vector_transport_to, v, x, y)
-    vector_transport_to!(M, vto, x, v, y)
+    vector_transport_to!(M, vto, x, v, y, m)
     return vto
 end
 
 """
-    vector_transport_direction!(M::Manifold, vto, x, v, vdir)
+    vector_transport_direction!(M::Manifold, vto, x, v, vdir[, m=::ParallelTransport])
```

Comment:
Should be `method=ParallelTransport()` or `method::AbstractVectorTransportMethod=ParallelTransport()`.